### ISSUE TITLE
Polygons with fill-extrusion follow the 3D Terrain

### ISSUE THREAD

**Issue by prebm-kng** [maplibre/maplibre-gl-js#3000]
Currently, when using fill-extrusion with 3D terrain enabled, the roof of an extruded feature is drawn as a flat plane. However, for certain applications, it might be useful for the elevated feature to follow the contour of the terrain, for instance, for vegetation coverage.

In the Mapbox-Issues, there has already been a discussion about this topic, exploring how it could be incorporated into the specifications: https://github.com/mapbox/mapbox-gl-js/issues/10355. However, I couldn't find a discussion related to this subject in the Maplibre issues.

Is such a functionality being considered or possible in Maplibre? I've taken a look at the code and noticed some differences from Mapbox's code. It seems that Maplibre has implemented this differently altogether. How complex would the implementation be and where would be some starting points?

**Comment by HarelM** [comment]
We can't copy code from mapbox, we shouldn't be looking at their code to avoid copyright infringement.
You are welcome to explore this, I would start with the style spec and a POC of the implementation if I were to try it.

**Comment by prebm-kng** [comment]
Thank you for your response. I am well aware of the copyright issues with mapbox and I wouldn't try to copy code from them. As part of my research I came over their issues and thats why I linked them. To my understanding their implementation is crucially different than maplibres implementation.

Thanks for the hints as where to search. I will close this issue for now and return if I have more questions.
